contracts/defi/provider/CompoundV2Provider.sol

```diff
@@ -0,0 +1,405 @@
+        exitMarketIfNeeded(_wallet, dToken, comptroller);
+    }
+
+    /**
+     * @dev Gets information about a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     * @return a status [0: no loan, 1: loan is safe, 2: loan is unsafe and can be liquidated] and the estimated ETH value of the loan
+     * combining all collaterals and all debts. When status = 1 it represents the value that could still be borrowed, while with status = 2
+     * it represents the value of collateral that should be added to avoid liquidation.      
+     */
+    function getLoan(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address[] calldata _oracles
+    ) 
+        external 
+        view 
+        returns (uint8 _status, uint256 _ethValue)
+    {
+        address comptroller = _oracles[0];
+        (uint error, uint liquidity, uint shortfall) = Comptroller(comptroller).getAccountLiquidity(address(_wallet));
+        require(error == 0, "Compound: failed to get account liquidity");
+        if(liquidity > 0) {
+            return (1, liquidity);
+        }
+        if(shortfall > 0) {
+            return (2, shortfall);
+        }
+        return (0,0);
+    }
+
+    /* ****************************************** Compound wrappers ******************************************* */
+
+    /**
+     * @dev Adds underlying tokens to a cToken contract.
+     * @param _wallet The target wallet.
+     * @param _cToken The cToken contract.
+     * @param _token The underlying token.
+     * @param _amount The amount of underlying token to add.
+     */
+    function mint(BaseWallet _wallet, address _cToken, address _token, uint256 _amount) internal {
+        require(_cToken != address(0), "Compound: No market for target token");
+        require(_amount > 0, "Compound: amount cannot be 0");
+        if(_token == ETH_TOKEN_ADDRESS) {
+            _wallet.invoke(_cToken, _amount, abi.encodeWithSignature("mint()"));
+        }
+        else {
+            _wallet.invoke(_token, 0, abi.encodeWithSignature("approve(address,uint256)", _cToken, _amount));
+            _wallet.invoke(_cToken, 0, abi.encodeWithSignature("mint(uint256)", _amount));
+        }
+    }
+
+    /**
+     * @dev Redeems underlying tokens from a cToken contract.
+     * @param _wallet The target wallet.
+     * @param _cToken The cToken contract.
+     * @param _amount The amount of cToken to redeem.
+     */
+    function redeem(BaseWallet _wallet, address _cToken, uint256 _amount) internal {     
+        require(_cToken != address(0), "Compound: No market for target token");   
+        require(_amount > 0, "Compound: amount cannot be 0");
+        _wallet.invoke(_cToken, 0, abi.encodeWithSignature("redeem(uint256)", _amount));
```

Comment:
Shall we maybe check the return value and revert on error?

Reply:
Same as above.